lib/Server.js

```diff
@@ -59,7 +58,16 @@ class Server {
     this.compiler = compiler;
     this.options = options;
 
-    handleStdin(options);
+    if (options.stdin) {
+      // listening for this event only once makes testing easier,
+      // since it prevents event listeners from hanging open
+      process.stdin.once('end', () => {
+        // eslint-disable-next-line no-process-exit
+        process.exit(0);
+      });
+
+      process.stdin.resume();
+    }
```

Comment:
No, i don't mean we don't need help, i just answer why we use this here?

Reply:
@evilebottnawi The goal is to make this option not CLI-only. As mentioned in https://github.com/webpack/webpack-dev-server/issues/1712 it is desired for this to be possible.

Also this is beneficial because I want to reduce what the CLI does as much as possible for the refactor.

Reply:
:+1: